third_party/ibis/ibis_DB2/compiler.py

```diff
@@ -363,6 +363,29 @@ def reduction_compiler(t, expr):
 
     return reduction_compiler
 
+def _count_start(sa_func):
+    return sa_func
+
+def _reduction_count(sa_func):
+    def formatter(t, expr):
+        op = expr.op()
+        *args, where = op.args
+
+        return _reduction_format(t, sa_func, where, *args)
+
+    return formatter
+
+def _reduction_format(t, sa_func, where, arg, *args):
+    if where is not None:
+        arg = t.translate(where.ifelse(arg, None))
+    else:
+        arg = t.translate(arg)
+
+    #Db2 doesn't allow '*' to be parameterized, probably better way to fix this... 
+    if arg == '*':
```

Comment:
Is is likely to break anything it touches?

I believe you could pull the column list from arg, but failing that perhaps it makes more sense to raise an exception

Reply:
It should have a fairly limited impact. It only runs for count functions and only if '*' is returned as a parameter. It is tested to work with multiple column counts.